website/addons/googledrive/tests/test_views.py

```diff
@@ -5,183 +5,80 @@
-        assert_is(self.node_settings.folder_path, None)
-        assert_is(self.node_settings.user_settings, None)
+from tests.factories import AuthUserFactory, ProjectFactory, ExternalAccountFactory
 
+from website.addons.googledrive.client import GoogleDriveClient
+from website.addons.googledrive.tests.utils import mock_folders as sample_folder_data
+from website.addons.googledrive.tests.factories import GoogleDriveAccountFactory
 
 class TestGoogleDriveConfigViews(OsfTestCase):
 
     def setUp(self):
         super(TestGoogleDriveConfigViews, self).setUp()
-        self.user = AuthUserFactory()
-        self.user.add_addon('googledrive')
-        self.user_settings = self.user.get_addon('googledrive')
+        self.account = GoogleDriveAccountFactory()
+        self.user = AuthUserFactory(external_accounts=[self.account])
+        self.user_settings = self.user.get_or_add_addon('googledrive')
         self.project = ProjectFactory(creator=self.user)
         self.project.add_addon('googledrive', Auth(self.user))
         self.node_settings = self.project.get_addon('googledrive')
         self.node_settings.user_settings = self.user_settings
-        oauth_settings = GoogleDriveOAuthSettingsFactory()
-        oauth_settings.save()
-        self.user_settings.oauth_settings = oauth_settings
         self.node_settings.save()
+        self.user_settings.save()
         # Log user in
         self.app.authenticate(*self.user.auth)
 
-    def test_drive_user_config_get_returns_correct_urls(self):
-        self.user.add_addon('googledrive')
-        url = api_url_for('googledrive_user_config_get', Auth(self.user))
+    def test_list_googledrive_accounts_returns_accounts_single(self):
+        url = api_url_for('list_googledrive_user_accounts', Auth(self.user))
         res = self.app.get(url)
         assert_equal(res.status_code, 200)
-        result = res.json['result']['urls']
-        assert_true(result['create'], api_url_for('googledrive_oauth_start_user'))
-        assert_true(result['delete'], api_url_for('googledrive_oauth_delete_user'))
+        assert_equal(len(res.json['accounts']), 1)
+        assert_equal(res.json['accounts'][0]['provider_id'], self.account.provider_id)
+        assert_equal(res.json['accounts'][0]['provider_name'], self.account.provider_name)
 
-    def test_drive_user_config_get_has_auth(self):
-        self.user.add_addon('googledrive')
-        user_settings = self.user.get_addon('googledrive')
-        user_settings.access_token = 'abc123'
-        user_settings.save()
-        url = api_url_for('googledrive_user_config_get', Auth(self.user))
+    def test_list_googledrive_accounts_returns_accounts_multiple(self):
+        external_account = GoogleDriveAccountFactory()
+        self.user.external_accounts.append(external_account) # self.account is already present
+        self.user.save()
+        url = api_url_for('list_googledrive_user_accounts', Auth(self.user))
```

Comment:
1. Since self.user is an AuthUserFactory instance, you should be able to just use `self.user.auth`. Except that...
2. `api_url_for` doesn't appear to have two positional arguments- can you please clarify the purpose of passing `Auth` in to `api_url_for`?